diff --git a/main/backtrace.c b/main/backtrace.c
index 2623d7ff87..9b80622b04 100644
--- a/main/backtrace.c
+++ b/main/backtrace.c
@@ -124,12 +124,12 @@ static void process_section(bfd *bfdobj, asection *section, void *obj)
 
 	offset = data->pc - (data->dynamic ? (bfd_vma)(uintptr_t) data->dli.dli_fbase : 0);
 
-	if (!(bfd_get_section_flags(bfdobj, section) & SEC_ALLOC)) {
+	if (!(bfd_section_flags(section) & SEC_ALLOC)) {
 		return;
 	}
 
-	vma = bfd_get_section_vma(bfdobj, section);
-	size = bfd_get_section_size(section);
+	vma = bfd_section_vma(section);
+	size = bfd_section_size(section);
 
 	if (offset < vma || offset >= vma + size) {
 		/* Not in this section */
